ocaml/typing/solver_intf.mli

```diff
@@ -118,21 +118,34 @@ module type Lattices_mono = sig
   val compose :
     'c obj -> ('b, 'c, 'd) morph -> ('a, 'b, 'd) morph -> ('a, 'c, 'd) morph
 
-  (* The following returns weaker than what we want, which is "\exists r.
-     allowed * r". But ocaml doesn't like existentials, and this weaker version
-     is good enough for us *)
-
-  (* Usual notion of adjunction f -| g :
-      [f x <= y] iff [x <= g y].
-
-     Our solver accepts a wider notion of adjunction and allows [f] and [g] to
-     be partial. They need to satisfy:
-     - If [f x <= y] and [y] is within [f]'s codomain, then [g] is well-defined
-     on [y] and [x <= g y].
-     - If not [f x <= y], we don't require anything about [g] on [y].
-     - If [y] is out of [f]'s codomain, we don't require anything about [g] on [y].
-
-     Similar in the other direction from [g] to [f]. *)
+  (* Usual notion of adjunction:
+     Given two morphisms [f : A -> B] and [g : B -> A], we require [f a <= b]
+      iff [a <= g b].
+
+     Our solver accepts a wider notion of adjunction and only requires the same
+     condition on convex sublattices. To be specific, if [f] and [g] form a
+      usual adjunction between [A] and [B], and [A] is a convex sublattice of
```

Comment:
A quick search for "convex sublattice" didn't find much immediately useful. Could you define this term in the comment?

Reply:
I found the definition here: https://en.wikipedia.org/wiki/Lattice_(order)#Sublattices